docs/usage/adding_awaits.rst

```diff
@@ -156,6 +156,118 @@ So, with that in mind, we can rewrite our example as the following:
 
 .. _return-value-callbacks:
 
+Using ``async with`` from C
+---------------------------
+Unlike the :c:func:`PyAwaitable_AddAwait` and the :c:func:`PyAwaitable_AddExpr`
+functions there exists :c:func:`PyAwaitable_AsyncWith` that will call the
+`__aenter__` and `__aexit__` members of a class if it implements the Async Context
```

Comment:
Should we add a reference link to the ``Async Context Manager`` in the CPython docs for those who might not know what it is?